File: 김효선/Sort/가장 큰수.py
from itertools import permutations

# def solution(numbers):
#     numbers = list(map(str, numbers))
    
#     arr = permutations(numbers)
#     arr = [''.join(p) for p in arr]
    
#     arr.sort()
#     answer=arr[-1]
    
#     return answer

def solution(numbers):

    numbers = list(map(str, numbers))
    numbers.sort(key=lambda x: x*3, reverse=True)
    answer = ''.join(numbers)
    
    return str(int(answer))

File: 이진영/Sort/가장 큰 수.js
function solution(numbers) {
    // 모든 수를 문자열로 변환하여 더하고, 그 값이 더 클 수록 앞으로 정렬
    let ret = numbers.map(i => i.toString()).sort((a, b) => (b + a) - (a + b)).join("");
    return ret[0] === '0' ? '0' : ret;
}